### PR TITLE
Implemented toast notification when new version of PWA is available and 'update' button to refresh on settings page (#85zrhdgkh)

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #

### Short Description and Why It's Useful

When a newer version of our app is available, a toast will be shown to notify the user and an update button will appear on the settings page to refresh the app.

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/import#contribution-guideline)

### CODE REVIEW

**Comment by k2maan** [comment]
The `refreshing` variable takes care of multiple refreshes. (Mentioned in the reference- https://dev.to/drbragg/handling-service-worker-updates-in-your-vue-pwa-1pip)

File: src/store/modules/user/actions.ts [hunk 1]
```diff
@@ -128,6 +128,17 @@ const actions: ActionTree<UserState, RootState> = {
     updateInstanceUrl(payload)
   },
 
+  updateFieldMappings({ commit }, payload){
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:131]: This is unrelated change

File: src/store/modules/user/actions.ts [hunk 2]
```diff
@@ -128,6 +128,17 @@ const actions: ActionTree<UserState, RootState> = {
...
+    commit(types.USER_FIELD_MAPPING_UPDATED, payload);
+  },
+
+  updatePwaRegistration({commit}, payload) {
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:135]: We could create a single action

File: src/store/modules/user/mutations.ts [hunk 3]
```diff
@@ -26,6 +26,12 @@ const mutations: MutationTree <UserState> = {
     [types.USER_DATETIME_FORMAT_UPDATED] (state, payload) {
         state.preferredDateTimeFormat = payload;
     },
+    [types.USER_PWA_REGISTRATION_UPDATED] (state, payload) {
```
Comment (adityasharma7) [inline src/store/modules/user/mutations.ts:29]: We could have single mutation for updating the pwaState

**Review by adityasharma7 — CHANGES_REQUESTED** [review]
